### DISCUSSION TITLE
Is macOS redirector limited to one read and one write from a connection ?

### DISCUSSION THREAD

**Discussion by charankamarapu** [mitmproxy/mitmproxy#7206] [Q&A] [answered]
From the code snippet https://github.com/mitmproxy/mitmproxy_rs/blob/15731b5e466e37da4aa164e30c1d98072c1c5b46/mitmproxy-macos/redirector/network-extension/TransparentProxyProvider.swift#L128 I can say that data is read from the client conn once and  data is written to client conn once. But what if the connection actually does more one read and write (like database connection)..?

**Comment by mhils** [comment, marked as answer]
See https://github.com/mitmproxy/mitmproxy_rs/blob/15731b5e466e37da4aa164e30c1d98072c1c5b46/mitmproxy-macos/redirector/network-extension/FlowExtensions.swift - it sets up two copiers that continuously forward data.

(You could have also just tried that out? :))
